flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamGraphGeneratorTest.java

```diff
@@ -960,6 +962,51 @@ public void testCacheSideOutput() {
         verifyCacheConsumeNode(env, upstreamParallelism, cacheTransformation);
     }
 
+    @Test
+    public void testTransformOutputEOFOperator() {
```

Comment:
It seems that this test aims to check "if an operator has isOutputOnEOF = true" and has keyed inputs, then its input should be sorted.

It is a bit weird/redundant to have this test since there is no logical connection between input requirement and isOutputOnEOF's value.

Note that it is in general impossible and unnecessary to test every combination of configurations. We typically only add tests whose input/output has logical connections.